Rules/AvoidOverwritingBuiltInCmdlets.cs

```diff
@@ -0,0 +1,284 @@
+                    {
+                        continue;
+                    }
+
+                    string functionName = funcDef.Name;
+                    foreach (var map in cmdletMap)
+                    {
+                        if (map.Value.Contains(functionName))
+                        {
+                            diagnosticRecords.Add(CreateDiagnosticRecord(functionName, map.Key, functionDef.Extent));
+                        }
+                    }
+                }
+
+                return diagnosticRecords;
+            }
+        }
+
+
+        private DiagnosticRecord CreateDiagnosticRecord(string FunctionName, string PSVer, IScriptExtent ViolationExtent)
+        {
+            var record = new DiagnosticRecord(
+                string.Format(CultureInfo.CurrentCulture,
+                    string.Format(Strings.AvoidOverwritingBuiltInCmdletsError, FunctionName, PSVer)),
+                ViolationExtent,
+                GetName(),
+                GetDiagnosticSeverity(),
+                ViolationExtent.File,
+                null
+            );
+            return record;
+        }
+
+
+        private void Initialize()
+        {
+            var psVerList = PowerShellVersion;
+
+            string settingsPath = Settings.GetShippedSettingsDirectory();
+
+            foreach (string reference in psVerList)
+            {
+                if (settingsPath == null || !ContainsReferenceFile(settingsPath, reference))
+                {
+                    return;
+                }
+            }
+
+            ProcessDirectory(settingsPath, psVerList);
+
+            if (cmdletMap.Keys.Count != psVerList.Count())
+            {
+                return;
+            }
+
+            initialized = true;
+        }
+
+
+        private bool ContainsReferenceFile(string directory, string reference)
+        {
+            return File.Exists(Path.Combine(directory, reference + ".json"));
+        }
+
+
+        private void ProcessDirectory(string path, IEnumerable<string> acceptablePlatformSpecs)
+        {
+            foreach (var filePath in Directory.EnumerateFiles(path))
+            {
+                var extension = Path.GetExtension(filePath);
+                if (String.IsNullOrWhiteSpace(extension)
+                    || !extension.Equals(".json", StringComparison.OrdinalIgnoreCase))
+                {
+                    continue;
+                }
+
+                var fileNameWithoutExt = Path.GetFileNameWithoutExtension(filePath);
+                if (acceptablePlatformSpecs != null
+                    && !acceptablePlatformSpecs.Contains(fileNameWithoutExt, StringComparer.OrdinalIgnoreCase))
+                {
+                    continue;
+                }
+
+                cmdletMap.Add(fileNameWithoutExt, GetCmdletsFromData(JObject.Parse(File.ReadAllText(filePath))));
+            }
+        }
+
+
+        private HashSet<string> GetCmdletsFromData(dynamic deserializedObject)
```

Comment:
This is the other disadvantage of the old profiles; they don't have a formalised schema. The more efficient way to do this would be to get the `JObject` instead though.

Reply:
@rjmholt do you recommend I change it? Perpetuating legacy practices isn't high on my to-do list, but I'm wondering how far removed from the desired practice it is.